src/cargo/util/dependency_queue.rs

```diff
@@ -149,15 +149,15 @@ impl<N: Hash + Eq + Clone, E: Eq + Hash + Clone, V> DependencyQueue<N, E, V> {
     ///
     /// A package is ready to be built when it has 0 un-built dependencies. If
     /// `None` is returned then no packages are ready to be built.
-    pub fn dequeue(&mut self) -> Option<(N, V)> {
-        let key = self
+    pub fn dequeue(&mut self) -> Option<(N, V, usize)> {
+        let (key, priority) = self
             .dep_map
             .iter()
             .filter(|(_, (deps, _))| deps.is_empty())
-            .map(|(key, _)| key.clone())
-            .max_by_key(|k| self.priority[k])?;
+            .map(|(key, _)| (key.clone(), self.priority[key]))
+            .max_by_key(|(_, priority)| *priority)?;
```

Comment:
This `max_by_key` is unnecessary as we are removing all and then sorting. If we change it to `.next()`, only the test will need to be updated.

Reply:
I can remove it if you want ? It doesn't seem to hurt today, and makes the order at least deterministic.

Reply:
I got nerd sniped, but it doesn't matter. Leave it bee.